imagebuildah/build.go

```diff
@@ -1817,13 +1817,20 @@ func (b *Executor) deleteSuccessfulIntermediateCtrs() error {
 
 func (s *StageExecutor) EnsureContainerPath(path string) error {
 	targetPath := filepath.Join(s.mountPoint, path)
-	_, err := os.Lstat(targetPath)
+	joinedPath, err := os.Lstat(targetPath)
 	if err != nil && os.IsNotExist(err) {
 		err = os.MkdirAll(targetPath, 0755)
 	}
 	if err != nil {
 		return errors.Wrapf(err, "error ensuring container path %q", path)
 	}
+	// If a symlink, check that the target directory exists in the container.
+	// If not, make it.
+	if joinedPath.Mode()&os.ModeSymlink != 0 {
+		if symTarget, err2 := os.Readlink(targetPath); err2 == nil {
+			os.MkdirAll(filepath.Join(s.mountPoint, symTarget), 0755)
```

Comment:
I believe you have to make sure that the path is withing the s.mountPoint,  Otherwise the link could create a directory on the host.

Reply:
libpod is using SecureJoin.
./vendor/github.com/cyphar/filepath-securejoin/join.go

Reply:
@rhatdan thanks for the pointer.  I was going through how to code that last night in my noggin, looks like it's already there with securejoin.